.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,23 @@
+name: Style Check
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
```

Comment:
I like the idea in general, is this a common practice nowadays? My only concern is that the main check is even more hidden inside layers of abstractions

But I agree that for each check we want to add is yet another action to configure (as an example back and clang format were different)

Reply:
It's true that it takes some getting used to. I first saw it used in some repo here on GH. It seemed interesting, so I started looking into it. And now it's used in all repos I maintain. No complaints so far, and actually at least one praise where a colleague said it's great that CI tells them exactly what to do. This is an example of a failure report in CI:

![image](https://user-images.githubusercontent.com/193852/228050260-383b50b8-4874-4d37-a4e6-d471e9ff96b7.png)

The syntax of the `.pre-commit-config.yaml` is a bit obscure, same with the GH actions (or `.gitlab-ci.yml` for that matter), but it's nothing that everyone must understand and know how to modify.

Reply:
Ok awesome, when I was wondering about the failure example was talking about the CI. So, they even tell you how to run it locally, this is the best of the world in my opinion. Usually people will come back with the error msg!

When I tried pre-commit 4 years ago it was not that developed, and thus I dropped it, it all seemed too chunky to maintain. But just putting it here to simplify the CI(that among all the examples you gave, the syntax that suck the most is the GH actions) it's more than worth it! I love the idea ;) Specially because pre-commit must NOT be enforced for developers (which was my prev intuition)

Reply:
Btw, the "If you are seeing this message in CI" is gone in newer versions apparently :cry: